File: sdks/java/src/main/java/com/puxbay/resources/AllResources.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.*;
import com.google.gson.reflect.TypeToken;
import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.Map;

/**
 * Base resource class with common HTTP methods.
 */
public abstract class BaseResource {
    protected final Puxbay client;
    
    protected BaseResource(Puxbay client) {
        this.client = client;
    }
    
    protected <T> T get(String endpoint, Class<T> responseType) throws PuxbayException {
        return client.request("GET", endpoint, null, responseType);
    }
    
    protected <T> T post(String endpoint, Object body, Class<T> responseType) throws PuxbayException {
        return client.request("POST", endpoint, body, responseType);
    }
    
    protected <T> T patch(String endpoint, Object body, Class<T> responseType) throws PuxbayException {
        return client.request("PATCH", endpoint, body, responseType);
    }
    
    protected void delete(String endpoint) throws PuxbayException {
        client.request("DELETE", endpoint, null, Void.class);
    }
}

/**
 * Products resource - Product management operations.
 */
public class ProductsResource extends BaseResource {
    public ProductsResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Product> list(int page, int pageSize) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<Product>>(){}.getType();
        return client.request("GET", String.format("products/?page=%d&page_size=%d", page, pageSize), null, type);
    }
    
    public Product get(String productId) throws PuxbayException {
        return get(String.format("products/%s/", productId), Product.class);
    }
    
    public Product create(Product product) throws PuxbayException {
        return post("products/", product, Product.class);
    }
    
    public Product update(String productId, Product product) throws PuxbayException {
        return patch(String.format("products/%s/", productId), product, Product.class);
    }
    
    public void delete(String productId) throws PuxbayException {
        delete(String.format("products/%s/", productId));
    }
    
    public Product adjustStock(String productId, int quantity, String reason) throws PuxbayException {
        Map<String, Object> body = new HashMap<>();
        body.put("quantity", quantity);
        body.put("reason", reason);
        return post(String.format("products/%s/stock-adjustment/", productId), body, Product.class);
    }
    
    public PaginatedResponse<ProductHistory> history(String productId, int page) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<ProductHistory>>(){}.getType();
        return client.request("GET", String.format("products/%s/history/?page=%d", productId, page), null, type);
    }
}

/**
 * Orders resource - Order management operations.
 */
public class OrdersResource extends BaseResource {
    public OrdersResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Order> list(int page, int pageSize) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<Order>>(){}.getType();
        return client.request("GET", String.format("orders/?page=%d&page_size=%d", page, pageSize), null, type);
    }
    
    public Order get(String orderId) throws PuxbayException {
        return get(String.format("orders/%s/", orderId), Order.class);
    }
    
    public Order create(Order order) throws PuxbayException {
        return post("orders/", order, Order.class);
    }
    
    public Order cancel(String orderId) throws PuxbayException {
        return post(String.format("orders/%s/cancel/", orderId), null, Order.class);
    }
}

/**
 * Customers resource - Customer management operations.
 */
public class CustomersResource extends BaseResource {
    public CustomersResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Customer> list(int page, int pageSize) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<Customer>>(){}.getType();
        return client.request("GET", String.format("customers/?page=%d&page_size=%d", page, pageSize), null, type);
    }
    
    public Customer get(String customerId) throws PuxbayException {
        return get(String.format("customers/%s/", customerId), Customer.class);
    }
    
    public Customer create(Customer customer) throws PuxbayException {
        return post("customers/", customer, Customer.class);
    }
    
    public Customer update(String customerId, Customer customer) throws PuxbayException {
        return patch(String.format("customers/%s/", customerId), customer, Customer.class);
    }
    
    public void delete(String customerId) throws PuxbayException {
        delete(String.format("customers/%s/", customerId));
    }
    
    public Customer adjustLoyaltyPoints(String customerId, int points, String description) throws PuxbayException {
        Map<String, Object> body = new HashMap<>();
        body.put("points", points);
        body.put("description", description);
        return post(String.format("customers/%s/adjust-loyalty-points/", customerId), body, Customer.class);
    }
    
    public Customer adjustStoreCredit(String customerId, double amount, String reference) throws PuxbayException {
        Map<String, Object> body = new HashMap<>();
        body.put("amount", amount);
        body.put("reference", reference);
        return post(String.format("customers/%s/adjust-store-credit/", customerId), body, Customer.class);
    }
}

/**
 * Categories resource - Category management operations.
 */
public class CategoriesResource extends BaseResource {
    public CategoriesResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Category> list(int page) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<Category>>(){}.getType();
        return client.request("GET", String.format("categories/?page=%d", page), null, type);
    }
    
    public Category get(String categoryId) throws PuxbayException {
        return get(String.format("categories/%s/", categoryId), Category.class);
    }
    
    public Category create(Category category) throws PuxbayException {
        return post("categories/", category, Category.class);
    }
    
    public Category update(String categoryId, Category category) throws PuxbayException {
        return patch(String.format("categories/%s/", categoryId), category, Category.class);
    }
    
    public void delete(String categoryId) throws PuxbayException {
        delete(String.format("categories/%s/", categoryId));
    }
}

/**
 * Suppliers resource - Supplier management operations.
 */
public class SuppliersResource extends BaseResource {
    public SuppliersResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Supplier> list(int page, int pageSize) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<Supplier>>(){}.getType();
        return client.request("GET", String.format("suppliers/?page=%d&page_size=%d", page, pageSize), null, type);
    }
    
    public Supplier get(String supplierId) throws PuxbayException {
        return get(String.format("suppliers/%s/", supplierId), Supplier.class);
    }
    
    public Supplier create(Supplier supplier) throws PuxbayException {
        return post("suppliers/", supplier, Supplier.class);
    }
    
    public Supplier update(String supplierId, Supplier supplier) throws PuxbayException {
        return patch(String.format("suppliers/%s/", supplierId), supplier, Supplier.class);
    }
    
    public void delete(String supplierId) throws PuxbayException {
        delete(String.format("suppliers/%s/", supplierId));
    }
}

/**
 * Gift Cards resource - Gift card operations.
 */
public class GiftCardsResource extends BaseResource {
    public GiftCardsResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<GiftCard> list(int page, String status) throws PuxbayException {
        String endpoint = String.format("gift-cards/?page=%d", page);
        if (status != null && !status.isEmpty()) {
            endpoint += "&status=" + status;
        }
        Type type = new TypeToken<PaginatedResponse<GiftCard>>(){}.getType();
        return client.request("GET", endpoint, null, type);
    }
    
    public GiftCard get(String cardId) throws PuxbayException {
        return get(String.format("gift-cards/%s/", cardId), GiftCard.class);
    }
    
    public GiftCard create(GiftCard card) throws PuxbayException {
        return post("gift-cards/", card, GiftCard.class);
    }
    
    public GiftCard redeem(String cardId, double amount) throws PuxbayException {
        Map<String, Object> body = new HashMap<>();
        body.put("amount", amount);
        return post(String.format("gift-cards/%s/redeem/", cardId), body, GiftCard.class);
    }
    
    public GiftCard checkBalance(String code) throws PuxbayException {
        return get(String.format("gift-cards/check-balance/?code=%s", code), GiftCard.class);
    }
}

/**
 * Branches resource - Branch management operations.
 */
public class BranchesResource extends BaseResource {
    public BranchesResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Branch> list(int page) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<Branch>>(){}.getType();
        return client.request("GET", String.format("branches/?page=%d", page), null, type);
    }
    
    public Branch get(String branchId) throws PuxbayException {
        return get(String.format("branches/%s/", branchId), Branch.class);
    }
    
    public Branch create(Branch branch) throws PuxbayException {
        return post("branches/", branch, Branch.class);
    }
    
    public Branch update(String branchId, Branch branch) throws PuxbayException {
        return patch(String.format("branches/%s/", branchId), branch, Branch.class);
    }
    
    public void delete(String branchId) throws PuxbayException {
        delete(String.format("branches/%s/", branchId));
    }
}

/**
 * Staff resource - Staff management operations.
 */
public class StaffResource extends BaseResource {
    public StaffResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Staff> list(int page, String role) throws PuxbayException {
        String endpoint = String.format("staff/?page=%d", page);
        if (role != null && !role.isEmpty()) {
            endpoint += "&role=" + role;
        }
        Type type = new TypeToken<PaginatedResponse<Staff>>(){}.getType();
        return client.request("GET", endpoint, null, type);
    }
    
    public Staff get(String staffId) throws PuxbayException {
        return get(String.format("staff/%s/", staffId), Staff.class);
    }
    
    public Staff create(Staff staff) throws PuxbayException {
        return post("staff/", staff, Staff.class);
    }
    
    public Staff update(String staffId, Staff staff) throws PuxbayException {
        return patch(String.format("staff/%s/", staffId), staff, Staff.class);
    }
    
    public void delete(String staffId) throws PuxbayException {
        delete(String.format("staff/%s/", staffId));
    }
}

/**
 * Webhooks resource - Webhook configuration operations.
 */
public class WebhooksResource extends BaseResource {
    public WebhooksResource(Puxbay client) {
        super(client);
    }
    
    public PaginatedResponse<Webhook> list(int page) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<Webhook>>(){}.getType();
        return client.request("GET", String.format("webhooks/?page=%d", page), null, type);
    }
    
    public Webhook get(String webhookId) throws PuxbayException {
        return get(String.format("webhooks/%s/", webhookId), Webhook.class);
    }
    
    public Webhook create(Webhook webhook) throws PuxbayException {
        return post("webhooks/", webhook, Webhook.class);
    }
    
    public Webhook update(String webhookId, Webhook webhook) throws PuxbayException {
        return patch(String.format("webhooks/%s/", webhookId), webhook, Webhook.class);
    }
    
    public void delete(String webhookId) throws PuxbayException {
        delete(String.format("webhooks/%s/", webhookId));
    }
    
    public PaginatedResponse<WebhookEvent> listEvents(String webhookId, int page) throws PuxbayException {
        Type type = new TypeToken<PaginatedResponse<WebhookEvent>>(){}.getType();
        return client.request("GET", String.format("webhook-logs/?webhook=%s&page=%d", webhookId, page), null, type);
    }
}

// Placeholder classes for remaining resources following same pattern
public class InventoryResource extends BaseResource {
    public InventoryResource(Puxbay client) { super(client); }
}

public class ReportsResource extends BaseResource {
    public ReportsResource(Puxbay client) { super(client); }
}

public class PurchaseOrdersResource extends BaseResource {
    public PurchaseOrdersResource(Puxbay client) { super(client); }
}

public class StockTransfersResource extends BaseResource {
    public StockTransfersResource(Puxbay client) { super(client); }
}

public class StocktakesResource extends BaseResource {
    public StocktakesResource(Puxbay client) { super(client); }
}

public class CashDrawersResource extends BaseResource {
    public CashDrawersResource(Puxbay client) { super(client); }
}

public class ExpensesResource extends BaseResource {
    public ExpensesResource(Puxbay client) { super(client); }
}

public class NotificationsResource extends BaseResource {
    public NotificationsResource(Puxbay client) { super(client); }
}

public class ReturnsResource extends BaseResource {
    public ReturnsResource(Puxbay client) { super(client); }
}

File: sdks/dotnet/src/Resources/AllResources.cs
using System.Net.Http;
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    /// <summary>
    /// Base resource class with common HTTP methods.
    /// </summary>
    public abstract class BaseResource
    {
        protected readonly PuxbayClient Client;
        
        protected BaseResource(PuxbayClient client)
        {
            Client = client;
        }
    }
    
    /// <summary>
    /// Products resource - Product management operations.
    /// </summary>
    public class ProductsResource : BaseResource
    {
        public ProductsResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Product>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await Client.RequestAsync<PaginatedResponse<Product>>(
                HttpMethod.Get, 
                $"products/?page={page}&page_size={pageSize}"
            );
        }
        
        public async Task<Product> GetAsync(string productId)
        {
            return await Client.RequestAsync<Product>(HttpMethod.Get, $"products/{productId}/");
        }
        
        public async Task<Product> CreateAsync(Product product)
        {
            return await Client.RequestAsync<Product>(HttpMethod.Post, "products/", product);
        }
        
        public async Task<Product> UpdateAsync(string productId, Product product)
        {
            return await Client.RequestAsync<Product>(HttpMethod.Patch, $"products/{productId}/", product);
        }
        
        public async Task DeleteAsync(string productId)
        {
            await Client.RequestAsync<object>(HttpMethod.Delete, $"products/{productId}/");
        }
        
        public async Task<Product> AdjustStockAsync(string productId, int quantity, string reason)
        {
            var body = new { quantity, reason };
            return await Client.RequestAsync<Product>(HttpMethod.Post, $"products/{productId}/stock-adjustment/", body);
        }
        
        public async Task<PaginatedResponse<ProductHistory>> HistoryAsync(string productId, int page = 1)
        {
            return await Client.RequestAsync<PaginatedResponse<ProductHistory>>(
                HttpMethod.Get, 
                $"products/{productId}/history/?page={page}"
            );
        }
    }
    
    /// <summary>
    /// Orders resource - Order management operations.
    /// </summary>
    public class OrdersResource : BaseResource
    {
        public OrdersResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Order>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await Client.RequestAsync<PaginatedResponse<Order>>(
                HttpMethod.Get, 
                $"orders/?page={page}&page_size={pageSize}"
            );
        }
        
        public async Task<Order> GetAsync(string orderId)
        {
            return await Client.RequestAsync<Order>(HttpMethod.Get, $"orders/{orderId}/");
        }
        
        public async Task<Order> CreateAsync(Order order)
        {
            return await Client.RequestAsync<Order>(HttpMethod.Post, "orders/", order);
        }
        
        public async Task<Order> CancelAsync(string orderId)
        {
            return await Client.RequestAsync<Order>(HttpMethod.Post, $"orders/{orderId}/cancel/");
        }
    }
    
    /// <summary>
    /// Customers resource - Customer management operations.
    /// </summary>
    public class CustomersResource : BaseResource
    {
        public CustomersResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Customer>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await Client.RequestAsync<PaginatedResponse<Customer>>(
                HttpMethod.Get, 
                $"customers/?page={page}&page_size={pageSize}"
            );
        }
        
        public async Task<Customer> GetAsync(string customerId)
        {
            return await Client.RequestAsync<Customer>(HttpMethod.Get, $"customers/{customerId}/");
        }
        
        public async Task<Customer> CreateAsync(Customer customer)
        {
            return await Client.RequestAsync<Customer>(HttpMethod.Post, "customers/", customer);
        }
        
        public async Task<Customer> UpdateAsync(string customerId, Customer customer)
        {
            return await Client.RequestAsync<Customer>(HttpMethod.Patch, $"customers/{customerId}/", customer);
        }
        
        public async Task DeleteAsync(string customerId)
        {
            await Client.RequestAsync<object>(HttpMethod.Delete, $"customers/{customerId}/");
        }
        
        public async Task<Customer> AdjustLoyaltyPointsAsync(string customerId, int points, string description)
        {
            var body = new { points, description };
            return await Client.RequestAsync<Customer>(HttpMethod.Post, $"customers/{customerId}/adjust-loyalty-points/", body);
        }
        
        public async Task<Customer> AdjustStoreCreditAsync(string customerId, decimal amount, string reference)
        {
            var body = new { amount, reference };
            return await Client.RequestAsync<Customer>(HttpMethod.Post, $"customers/{customerId}/adjust-store-credit/", body);
        }
    }
    
    /// <summary>
    /// Categories resource - Category management operations.
    /// </summary>
    public class CategoriesResource : BaseResource
    {
        public CategoriesResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Category>> ListAsync(int page = 1)
        {
            return await Client.RequestAsync<PaginatedResponse<Category>>(HttpMethod.Get, $"categories/?page={page}");
        }
        
        public async Task<Category> GetAsync(string categoryId)
        {
            return await Client.RequestAsync<Category>(HttpMethod.Get, $"categories/{categoryId}/");
        }
        
        public async Task<Category> CreateAsync(Category category)
        {
            return await Client.RequestAsync<Category>(HttpMethod.Post, "categories/", category);
        }
        
        public async Task<Category> UpdateAsync(string categoryId, Category category)
        {
            return await Client.RequestAsync<Category>(HttpMethod.Patch, $"categories/{categoryId}/", category);
        }
        
        public async Task DeleteAsync(string categoryId)
        {
            await Client.RequestAsync<object>(HttpMethod.Delete, $"categories/{categoryId}/");
        }
    }
    
    /// <summary>
    /// Suppliers resource - Supplier management operations.
    /// </summary>
    public class SuppliersResource : BaseResource
    {
        public SuppliersResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Supplier>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await Client.RequestAsync<PaginatedResponse<Supplier>>(
                HttpMethod.Get, 
                $"suppliers/?page={page}&page_size={pageSize}"
            );
        }
        
        public async Task<Supplier> GetAsync(string supplierId)
        {
            return await Client.RequestAsync<Supplier>(HttpMethod.Get, $"suppliers/{supplierId}/");
        }
        
        public async Task<Supplier> CreateAsync(Supplier supplier)
        {
            return await Client.RequestAsync<Supplier>(HttpMethod.Post, "suppliers/", supplier);
        }
        
        public async Task<Supplier> UpdateAsync(string supplierId, Supplier supplier)
        {
            return await Client.RequestAsync<Supplier>(HttpMethod.Patch, $"suppliers/{supplierId}/", supplier);
        }
        
        public async Task DeleteAsync(string supplierId)
        {
            await Client.RequestAsync<object>(HttpMethod.Delete, $"suppliers/{supplierId}/");
        }
    }
    
    /// <summary>
    /// Gift Cards resource - Gift card operations.
    /// </summary>
    public class GiftCardsResource : BaseResource
    {
        public GiftCardsResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<GiftCard>> ListAsync(int page = 1, string status = null)
        {
            var endpoint = $"gift-cards/?page={page}";
            if (!string.IsNullOrEmpty(status))
            {
                endpoint += $"&status={status}";
            }
            return await Client.RequestAsync<PaginatedResponse<GiftCard>>(HttpMethod.Get, endpoint);
        }
        
        public async Task<GiftCard> GetAsync(string cardId)
        {
            return await Client.RequestAsync<GiftCard>(HttpMethod.Get, $"gift-cards/{cardId}/");
        }
        
        public async Task<GiftCard> CreateAsync(GiftCard card)
        {
            return await Client.RequestAsync<GiftCard>(HttpMethod.Post, "gift-cards/", card);
        }
        
        public async Task<GiftCard> RedeemAsync(string cardId, decimal amount)
        {
            var body = new { amount };
            return await Client.RequestAsync<GiftCard>(HttpMethod.Post, $"gift-cards/{cardId}/redeem/", body);
        }
        
        public async Task<GiftCard> CheckBalanceAsync(string code)
        {
            return await Client.RequestAsync<GiftCard>(HttpMethod.Get, $"gift-cards/check-balance/?code={code}");
        }
    }
    
    /// <summary>
    /// Branches resource - Branch management operations.
    /// </summary>
    public class BranchesResource : BaseResource
    {
        public BranchesResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Branch>> ListAsync(int page = 1)
        {
            return await Client.RequestAsync<PaginatedResponse<Branch>>(HttpMethod.Get, $"branches/?page={page}");
        }
        
        public async Task<Branch> GetAsync(string branchId)
        {
            return await Client.RequestAsync<Branch>(HttpMethod.Get, $"branches/{branchId}/");
        }
        
        public async Task<Branch> CreateAsync(Branch branch)
        {
            return await Client.RequestAsync<Branch>(HttpMethod.Post, "branches/", branch);
        }
        
        public async Task<Branch> UpdateAsync(string branchId, Branch branch)
        {
            return await Client.RequestAsync<Branch>(HttpMethod.Patch, $"branches/{branchId}/", branch);
        }
        
        public async Task DeleteAsync(string branchId)
        {
            await Client.RequestAsync<object>(HttpMethod.Delete, $"branches/{branchId}/");
        }
    }
    
    /// <summary>
    /// Staff resource - Staff management operations.
    /// </summary>
    public class StaffResource : BaseResource
    {
        public StaffResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Staff>> ListAsync(int page = 1, string role = null)
        {
            var endpoint = $"staff/?page={page}";
            if (!string.IsNullOrEmpty(role))
            {
                endpoint += $"&role={role}";
            }
            return await Client.RequestAsync<PaginatedResponse<Staff>>(HttpMethod.Get, endpoint);
        }
        
        public async Task<Staff> GetAsync(string staffId)
        {
            return await Client.RequestAsync<Staff>(HttpMethod.Get, $"staff/{staffId}/");
        }
        
        public async Task<Staff> CreateAsync(Staff staff)
        {
            return await Client.RequestAsync<Staff>(HttpMethod.Post, "staff/", staff);
        }
        
        public async Task<Staff> UpdateAsync(string staffId, Staff staff)
        {
            return await Client.RequestAsync<Staff>(HttpMethod.Patch, $"staff/{staffId}/", staff);
        }
        
        public async Task DeleteAsync(string staffId)
        {
            await Client.RequestAsync<object>(HttpMethod.Delete, $"staff/{staffId}/");
        }
    }
    
    /// <summary>
    /// Webhooks resource - Webhook configuration operations.
    /// </summary>
    public class WebhooksResource : BaseResource
    {
        public WebhooksResource(PuxbayClient client) : base(client) { }
        
        public async Task<PaginatedResponse<Webhook>> ListAsync(int page = 1)
        {
            return await Client.RequestAsync<PaginatedResponse<Webhook>>(HttpMethod.Get, $"webhooks/?page={page}");
        }
        
        public async Task<Webhook> GetAsync(string webhookId)
        {
            return await Client.RequestAsync<Webhook>(HttpMethod.Get, $"webhooks/{webhookId}/");
        }
        
        public async Task<Webhook> CreateAsync(Webhook webhook)
        {
            return await Client.RequestAsync<Webhook>(HttpMethod.Post, "webhooks/", webhook);
        }
        
        public async Task<Webhook> UpdateAsync(string webhookId, Webhook webhook)
        {
            return await Client.RequestAsync<Webhook>(HttpMethod.Patch, $"webhooks/{webhookId}/", webhook);
        }
        
        public async Task DeleteAsync(string webhookId)
        {
            await Client.RequestAsync<object>(HttpMethod.Delete, $"webhooks/{webhookId}/");
        }
        
        public async Task<PaginatedResponse<WebhookEvent>> ListEventsAsync(string webhookId, int page = 1)
        {
            return await Client.RequestAsync<PaginatedResponse<WebhookEvent>>(
                HttpMethod.Get, 
                $"webhook-logs/?webhook={webhookId}&page={page}"
            );
        }
    }
    
    // Placeholder classes for remaining resources following same pattern
    public class InventoryResource : BaseResource
    {
        public InventoryResource(PuxbayClient client) : base(client) { }
    }
    
    public class ReportsResource : BaseResource
    {
        public ReportsResource(PuxbayClient client) : base(client) { }
    }
    
    public class PurchaseOrdersResource : BaseResource
    {
        public PurchaseOrdersResource(PuxbayClient client) : base(client) { }
    }
    
    public class StockTransfersResource : BaseResource
    {
        public StockTransfersResource(PuxbayClient client) : base(client) { }
    }
    
    public class StocktakesResource : BaseResource
    {
        public StocktakesResource(PuxbayClient client) : base(client) { }
    }
    
    public class CashDrawersResource : BaseResource
    {
        public CashDrawersResource(PuxbayClient client) : base(client) { }
    }
    
    public class ExpensesResource : BaseResource
    {
        public ExpensesResource(PuxbayClient client) : base(client) { }
    }
    
    public class NotificationsResource : BaseResource
    {
        public NotificationsResource(PuxbayClient client) : base(client) { }
    }
    
    public class ReturnsResource : BaseResource
    {
        public ReturnsResource(PuxbayClient client) : base(client) { }
    }
}
